Requested updates from Flickerbox

chrislarrycarl opened the pull request:
Hey John,

This pull request includes the following requested changes:

> It takes too long to render the TOC

Add "loading" state earlier in the page rendering to avoid content shifting and add demarcation for where the TOC will load

> TOC can not be opened on "mobile" mode, only first level items can be visited

Change docs.js to fully expand TOC on mobile by default, allowing full access to items (can be adjusted by changing selector in docs.js line 312)

> Header is too tall on "mobile" view

Change header from 100px to 75px and reduce menu button height on screens < medium

> Search button doesn't work on mobile view

Change link to http://developer.okta.com/search/

> In expanded view, the grey color on sidebar should extend fully to the left, top, and bottom of the page

Change .Sidebar styles to extend fully to the left, top, and bottom of the page

> When the level1 TOC content wraps to a second line, it interferes with the line that is displayed to the left of the selected level 2 content.

Change docs.js to add a bit of cushion for titles > 50 characters

> One some pages, SOME of the the TOC content disappears when you scroll to the bottom: 

The preloader now uses the page title as the title for all tables of contents, this pushes the page h1 "lower" in the TOC to be equal with the page h2s. This makes the TOC title more context relevant (instead of showing "Overview" the title is now "Sessions" for example - http://developer.okta.com.dev/docs/api/resources/sessions.html), this additionally fixes the issue where a page with fewer headings loses items when scrolling (a page with a single H1 and only two child H2s as in the api access management page for example).

> The "documentation" page doesn't render properly on IE 11

Fix for grid layout for IE to constrain flex width and address padding expansion of element width

> Left Nav Style/Behavior Fixes
- Fix Top level titles 400 weight, all caps, underline
- Fix categories style match to top level pages
- Fix remove left border for secondary pages (nested under categories)
- Fix collapse inactive categories, expand active category only
- Fix active page highlighting consistent without regard for page depth

Additional Included Change:

The master.js file was being included twice on the code pages causing the support dropdown to be disabled (http://developer.okta.com/code/php/). Second include has been removed from _layouts/software.html to resolve the issue.

Thanks!,
-Chris

johngronberg-okta commented:
Thanks @chrislarrycarl We'll do some testing and code review here and see how it looks.

@jpf can you help take a look at this today?

johngronberg-okta commented:
A couple issues I've noticed:

- Clicking Authentication Reference, API Reference, or Getting Started in the Left Nav navigates to an empty page. This wasn’t an issue before because those nav items weren’t clickable. For now, just open whatever is the first page listed as a subpage, so
Getting Started -> API Test Client
Authentication Reference -> Authentication API
API reference -> Apps API

- On /documentation we now need to add another link:
_Authentication Reference: Documentation describing how to use Okta APIs to authenticate users. Includes sample requests and responses._

- On /documentation, API reference should direct to the API reference page with Apps API automatically selected. Right now it goes to Authentication Reference.

chrislarrycarl commented:
Hey @johngronberg-okta,

I've updated the pull request with changes to address these items. 

- Add _reference/authentication_reference.md file to include section in /documentation
- Change mobile table of contents to h4 styling from h2
- Change category links in sidebar to first child page for categories (redirect files unaltered)
- Change _reference/*.md links so sections in /documentation link directly to child page
- Fix link in _/reference/api_reference.md

Thanks!,
-Chris

mystiberry-okta commented:
@johngronberg-okta or @joelfranusic-okta  why did Flickerbox change the doc title from Platform Release Notes to Release Notes? We need that everywhere to help disambiguate them from Product Release Notes.

johngronberg-okta commented on _source/_docs/platform-release-notes/platform-release-notes.md:
```diff
@@ -1,6 +1,6 @@
 ---
 layout: docs_page
-title: Platform Release Notes
+title: Release Notes
```
This should remain as "Platform Release Notes"

chrislarrycarl commented:
Hi @johngronberg-okta,

I've reverted the title to "Platform Release Notes". To enable active nav highlighting document.title and page.title must match so I've standardized "Platform Release Notes" within the other files. In addition to enabling active nav highlighting this improves consistency across the site, making the link text in the /documentation Reference section as well as the sidebar nav match the preferred verbiage.

Additionally, I adjusted the active nav trigger in the sidebar for the bottom block of links so that historical Product Release Notes can utilize unique page titles while still properly activating the nav highlighting. Moving forward these pages will just need to include "Product Release Notes" somewhere in their title. This will allow updating the recent historical release notes to utilize unique page titles consistent with the older pages.

Thanks!
-Chris

mystiberry-okta commented:
Not sure what you're talking about as there are no Product Release Notes on developer.okta.com. Did you mean Platform? In any case, we don't want the historical ones listed--it'd be a list of over 50 documents, not what we want displayed in the left nav.

chrislarrycarl commented:
Sorry @mystiberry-okta that was a typo. It should have read "Platform Release Notes", I've updated my original comment.

mystiberry-okta commented:
sounds good. Thanks!

mystiberry-okta commented:
@johngronberg-okta  Waiting for your okay to merge (and republish?).

joelfranusic-okta commented:
Hi @chrislarrycarl - I'm unable to scroll pages on iOS Safari [1]. For example, if I navigate to `/docs/api/resources/authn.html`, I am not able to scroll the page when I start the scroll inside of the Table of Contents area. In the image below, I am unable to scroll inside of the red highlighted area, but I am able to scroll inside of the green highlighted area: 
![img_0308](https://cloud.githubusercontent.com/assets/8584286/20582070/74acde9c-b192-11e6-8651-73f425c9e3ac.png)

Footnote:
1: `Mozilla/5.0 (iPhone; CPU iPhone OS 10_1_1 like Mac OS X) AppleWebKit/602.2.14 (KHTML, like Gecko) Version/10.0 Mobile/14B100 Safari/602.1`

chrislarrycarl commented:
Hi @joelfranusic-okta, 

The change I just pushed should resolve this issue. If it does not please let me know.

Thanks!,
-Chris

joelfranusic-okta commented:
That fixed it, thanks @chrislarrycarl!

mystiberry-okta commented:
@joelfranusic-okta or @johngronberg-okta  Should I merge this now?

johngronberg-okta commented:
I've asked a couple times for others to test and verify these changes. Before we merge I really need another set of eyes on this PR


________________________________
From: Mysti Berry <notifications@github.com>
Sent: Wednesday, November 30, 2016 8:36:36 AM
To: okta/okta.github.io
Cc: John Gronberg; Mention
Subject: Re: [okta/okta.github.io] Requested updates from Flickerbox (#500)


@joelfranusic-okta<https://github.com/joelfranusic-okta> or @johngronberg-okta<https://github.com/johngronberg-okta> Should I merge this now?

-
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/okta/okta.github.io/pull/500#issuecomment-263922945>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AVdSG6CZH50OZ2R9gAJhdCf-2iI_vR5fks5rDaYUgaJpZM4K2woc>.

chrislarrycarl commented:
Hi @joelfranusic-okta, @mystiberry-okta, & @johngronberg-okta 

I sneaked an additional fix into this pull request for a typo on the homepage.

Thanks!

rchild-okta commented on _source/assets/js/docs.js:
```diff
@@ -77,7 +78,7 @@ $(function() {
 		var $content   = $('.PageContent-main');
 		var $headers   = [];
-		var $toc       = $('<div>').addClass('TableOfContents').appendTo($page);
+		var $toc       = ($('.TableOfContents').length > 0) ? $('.TableOfContents') : $('<div>').addClass('TableOfContents').appendTo($page);
```
Curious - what are we protecting against here? When do we have a table of contents already on the page (i.e. isn't this the part of the code that builds it)?

rchild-okta commented on _source/assets/js/docs.js:
```diff
@@ -105,14 +108,19 @@ $(function() {
 					.appendTo($toc);
 			}
 
+			if (document.title.length > 50) {
```
Can you explain this? Is the idea that we expect a title of length > 50 to drop down to another line, so we need to change the look?

chrislarrycarl replied:
Yes, to accommodate pages with varying length titles a limit of 50 was added to increase spacing and allow for wrap. These style changes have been moved to a class in the sass file for consistency.

rchild-okta commented on _source/assets/js/docs.js:
```diff
@@ -247,57 +239,83 @@ $(function() {
+			else {
+				// comment out to keep mobile TOC limited to level1 items visible only
+				// or change selector to display/hide varying levels of depth
+				$('.is-level2, .is-level3, .is-level4, .is-level5, .is-level6').css({'display':'block'});
```
Instead of doing it through JS, can we do this through CSS and applying a parent class? I.e. parent class of 'is-mobile', and then the child elements are display:block.

mystiberry-okta commented:
@johngronberg-okta Are we still waiting for Flickerbox to address Royal's comments?

johngronberg-okta commented:
Yes- We need Reman's comments to be resolved.

chrislarrycarl commented on _source/_includes/sidebar.html:
```diff
@@ -1,105 +1,134 @@
+            {% assign documents = site.use_cases | sort: 'weight' %}
+            {% for document in documents %}
+            {% unless document.id contains '/index' %}
+            {% continue %}
```
That was an oversight on my part, continue indentation has been fixed.

chrislarrycarl commented on _source/_includes/sidebar.html:
```diff
@@ -1,105 +1,134 @@
+            {% endunless %}
+            <li{% if document.title == page.title%} class="is-active"{% endif %}><a href="{{ document.url | remove: '/index' }}">{{ document.title }}</a></li>
+            {% endfor %}
+            <!-- <li class="is-active"> -->
```
All commented code has been removed from the sidebar.html file.

chrislarrycarl commented:
Hi @rchild-okta, @mystiberry-okta, @johngronberg-okta,

I believe the outstanding items above have been addressed. Please let me know if there's anything else I can do to assist.

Thanks!

mystiberry-okta commented:
@johngronberg-okta are we okay to merge & publish today?

johngronberg-okta commented:
That's up to @rchild-okta based on chris' replies to the outstanding comments.

johngronberg-okta commented:
Hey Chris,

Sorry, an issue I found last minute while testing. I can't scroll on pages in the mobile view with the TOC expanded. You can re-create the issue easily on /docs/api/resources/authn.html#overview

We fixed this in a previous PR., but I think it's something going on with the scroll lock library and we need to disable it on mobile for the TOC

John

johngronberg-okta commented:
Actually, nevermind @chrislarrycarl I don't think the issue is related to your changes. Sorry for the false alarm!

chrislarrycarl commented:
@johngronberg-okta The change from commit a14db89 should have fixed the scroll lock when loading on mobile, however, in investigating this just now I did see that when on desktop and sizing down to mobile the lock persists. I'll send a new pull request with that fix.
